.github/workflows/tests.yml

```diff
@@ -38,18 +40,30 @@ jobs:
   build-snap:
     name: Build snap
     runs-on: ubuntu-latest
+    if: ${{ !github.event.pull_request.draft }}
     needs: lint-unittests-docs
+    outputs:
+      files-changed: ${{ steps.files-changed.outputs.changed }}
     steps:
     - uses: actions/checkout@v2
     - uses: snapcore/action-build@v1
       id: snapcraft
     - name: Install snap
       run: sudo snap install --dangerous ${{ steps.snapcraft.outputs.snap }}
+    - name: Track changed files in the PR
+      id: files-changed
+      run: |
+        git remote add upstream https://github.com/canonical/juju-verify
+        git fetch upstream master
+        failed=0
+        git diff --name-only upstream/master | grep -qE '^(juju_verify|tests)' || failed=1
+        echo "::set-output name=changed::$failed"
 
   func-tests:
     name: Integration tests
     runs-on: self-hosted
     needs: build-snap
+    if: ${{ !github.event.pull_request.draft && needs.build-snap.outputs.files-changed == 0 }}
```

Comment:
Can we avoid using Draft PR because PR could not be change back to Draft?
Maybe we can try to use `github.event.pull_request.review_requested` as I did in #20?
How I think it should work:
 - you will request a review from someone -> trigger build snap and integration tests
 - the person will give you a review with "requested changes"
 - you will push your changes (one or multiple commits) and merge master to PR (if needed)
 - you will trigger "re-request review" -> trigger build snap and integration tests

Reply:
We may want to request reviews from colleagues when the PR is still a draft (like, if we want to discuss a certain approach/logic in the code).

Reply:
Yes, that good point. We can use `github.event.pull_request.review_requested` and `!github.event.pull_request.draft`.
I can see the problem with this approach mainly when someone has applied suggestion from review and pushed them, but then he/she realized that master need to be merged into this branch due some conflict.

I would also like to use the "re-request review" button to alert the reviewer that their review is needed again.